README.md

```diff
@@ -310,6 +325,54 @@ function assignId (req, res, next) {
 }
 ```
 
+#### arguments to custom token formats
+
+Example of a custom token format that uses an argument. It will echo the
+argument passed to the token.
+
+```js
+var express = require('express')
+var morgan = require('morgan')
+
+morgan.token('echo', function getId (req, res, arg) {
+  return arg
+})
+
+var app = express()
+
+app.use(morgan(':echo[hello world!] :response-time'))
+
+app.get('/', function (req, res) {
+  res.send('hello, world!')
+})
+```
+
+### use custom format function
+
+Sample app that will use a custom formatting function. This will print JSON
+instead of a simple string.
+
+```js
+var express = require('express')
+var morgan = require('morgan')
+
+var combined = morgan.compile('[:date[clf]] :method :url :res[content-length]')
+var JSONOutput = function (morgan, req, res) {
+  return JSON.stringify({
+    status: morgan.status(req,res),
+    default: combined(morgan, req, res),
```

Comment:
Is it possible to use another variable than `combined` since it's not the combined format? Also, can you use a named function instead of an anonymous function? Or you can use an anonymous function directly in the `morgan()` call.